packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -73,19 +74,33 @@ function isHexString(v: unknown): v is string {
   return v.trim().match(HEX_STRING_REGEX) !== null;
 }
 
+function isDecString(v: unknown): v is string {
+  if (typeof v !== "string") {
+    return false;
+  }
+
+  return v.match(DEC_STRING_REGEX) !== null;
+}
+
 export const hexString = new t.Type<string>(
   "hex string",
   isHexString,
   (u, c) => (isHexString(u) ? t.success(u) : t.failure(u, c)),
   t.identity
 );
 
+export const decString = new t.Type<string>(
```

Comment:
I would replace all `dec` for `decimal`, to make it more clear. I know we use `hex` instead of `hexadecimal`, but `hex` is easier to understand that `dec` IMO.